plugins/binaryninja/tests/test_native_plugin.py

```diff
@@ -142,5 +143,201 @@ def test_global(self) -> None:
+        4198568,
+        4198576,
+        4198632,
+        4198640,
+        4198653,
+        4198679,
+        4198684,
+        4198704,
+        4198713,
+        4198733,
+        4198784,
+        4198836,
+        4198848,
+        4198862,
+        4198896,
+        4198902,
+        4198910,
+        4198923,
+        4198963,
+        4198979,
+        4199006,
+        4199016,
+        4199023,
+        4199034,
+        4199042,
+        4199055,
+        4199181,
+        4199192,
+        4199214,
+        4199229,
+        4199328,
+        4199343,
+        4199362,
+        4199370,
+        4199379,
+        4199392,
+        4199428,
+        4199438,
+        4199456,
+        4199490,
+        4199520,
+        4199536,
+        4199565,
+        4199576,
+        4199581,
+        4199600,
+        4199621,
+        4199641,
+        4199645,
+        4199687,
+        4199703,
+        4199712,
+        4199729,
+        4199808,
+        4199824,
+        4199828,
+        4199830,
+        4199861,
+        4199936,
+        4200043,
+        4200096,
+        4200123,
+        4200160,
+        4200172,
+        4200181,
+        4200192,
+        4200225,
+        4200236,
+        4200307,
+        4200350,
+        4200442,
+        4200447,
+        4200455,
+        4200460,
+        4200484,
+        4200496,
+        4200544,
+        4200576,
+        4200912,
+        4200954,
+        4200963,
+        4200976,
+        4200992,
+        4200998,
+        4201019,
+        4201168,
+        4201217,
+        4201236,
+        4201241,
+        4201255,
+        4201271,
+        4201312,
+        4201339,
+        4201424,
+        4201443,
+        4201465,
+        4201488,
+        4201524,
+        4201537,
+        4201552,
+        4201563,
+        4201584,
+        4201595,
+        4201598,
+        4201603,
+        4201696,
+        4201704,
+        4201708,
+        4201744,
+        4201803,
+        4201814,
+        4201833,
+        4201904,
+        4201917,
+        4201944,
+        4201982,
+        4202007,
+        4202016,
+        4202020,
+        4202025,
+        4202029,
+        4202038,
+        4202058,
+        4202073,
+        4202078,
+        4202079,
+        4202094,
+        4202128,
+        4202138,
+        4202161,
+        4202224,
+        4202237,
+        4202288,
+        4202293,
+        4202307,
+        4202327,
+        4202384,
+        4202392,
+        4202400,
+        4202410,
+        4202441,
+        4202453,
+        4202465,
+        4202480,
+        4202633,
+        4202648,
+        4202656,
+        4202661,
+        4202705,
+        4202784,
+        4202804,
+        4202832,
+        4202912,
+        4203138,
+    }
+    # Init libc seems to behave differently between manticore and native, ignore blocks from it
+    INIT_LIBC = 0x401180
+    INIT_LIBC_END = 0x401398
+    # IOCTL for fd={0,1,2} on manticore currently doesn't work
```

Comment:
Do you know anything more about what "doesn't work" in Manticore?

Reply:
I think "doesn't work" might've been the wrong word choice. But here is the code explainer:

```py
# manticore/platforms/linux.py:1039
    def _init_std_fds(self) -> None:
        ...
        stdin = Socket()
        stdout = Socket()
        stderr = Socket()
        ...
        in_fd = self._open(stdin)
        out_fd = self._open(stdout)
        err_fd = self._open(stderr)
        ...

# manticore/platforms/linux.py:726
@concreteclass
class Socket(FdLike):
    ...
    def ioctl(self, request, argp):
        raise FdError("Invalid ioctl() operation on Socket", errno.ENOTTY)
```

IOCTL on fd=0,1,2 will always return an `ENOTTY` which would be different from the behaviour when running in the terminal.

Reply:
Actually, maybe I'll try to regenerate the reference drcov trace but without a TTY.

Reply:
https://github.com/trailofbits/ManticoreUI/pull/100/commits/26925f5366a83f4ea28746a863f95e79b50af548 fixes this

Reply:
Ahhh interesting... I think we should mention this in the source code comment because it's not obvious that the redirection is required for correct operation. Thank you for investigating!